tests/components/test_alert.py

```diff
@@ -170,3 +197,12 @@ def test_noack(self):
         self.hass.block_till_done()
 
         self.assertEqual(True, entity.hidden)
+
+    def test_done_message_state_tracker_reset_on_cancel(self):
+        entity = alert.Alert(self.hass, *TEST_NOACK)
+        entity._cancel = lambda *args: None
+        assert entity._send_done_message == False
+        entity._send_done_message = True
+        self.hass.add_job(entity.end_alerting)
+        self.hass.block_till_done()
+        assert entity._send_done_message == False
```

Comment:
nice catch.  I've submitted the change with a unit test.  If you think it'll fly, sweet, if not, please give me a brief pointer as to a more correct approach.